docs/topics/native/native-spm.md

```diff
@@ -0,0 +1,194 @@
+call to your iOS targets description in the `shared/build.gradle.kts` file:
+   ```kotlin
+   import org.jetbrains.kotlin.gradle.plugin.mpp.apple.XCFramework
+   
+   kotlin {
+      // other KMP targets
+      // ...
+      val xcf = XCFramework()
+      // Name of the module that will be imported in the consumer project
+      val xcframeworkName = "ExampleDependency"
+   
+      listOf(
+         iosX64(),
+         iosArm64(),
+         iosSimulatorArm64(),
+      ).forEach {
+           it.binaries.framework {
+              baseName = xcframeworkName
+              xcf.add(this)
+           }
+        }
+      //...
+   }
+   ```
+2. Sync the updated Gradle configuration.
+3. Run the Gradle task to create the framework:
+   
+   `./gradlew :shared:assembleSharedReleaseXCFramework`
+4. The resulting framework will be created as the `shared/build/XCFrameworks/release/shared.xcframework` folder in your project directory.
+5. Put the `shared.xcframework` folder in a ZIP archive and calculate the checksum for the resulting archive, for example:
+   
+   `swift package compute-checksum shared.xcframework.zip`
+6. Upload the ZIP file to the file storage of your choice.
+7. Create a `Package.swift` file with the following code:
+   ```Swift
+   // swift-tools-version:5.3
+   import PackageDescription
+    
+   let package = Package(
+      name: "shared",
+      platforms: [
+        .iOS(.v14),
+      ],
+      products: [
+         .library(name: "shared", targets: ["shared"])
+      ],
+      targets: [
+         .binaryTarget(
+            name: "shared",
+            url: "<link to the uploaded XCFramework ZIP file>"),
+            checksum:"<checksum calculated for the ZIP file>"
+      ]
+   )
+   ```
+8. You can check if the package manifest is valid by running the following command next to the `Package.swift` file:
+
+    ```shell
+    swift package describe
+    ```
+    
+    The output will describe any found errors, or show the parsed tree of the manifest if it is grammatically correct. 
+9. Push the `Package.swift` file to the repository you settled on earlier. Make sure to create a git tag with the
+Semantic Version of the package.
+
+Now that both files are accessible, you can try and set up the dependency in Xcode:
+* In an Xcode project, choose **File | Add Package Dependencies...** and provide the Git URL for the repo with
+the `Package.swift` file.
+* In an SPM project, create a dependency following the Apple documentation on [Package.Dependency](https://developer.apple.com/documentation/packagedescription/package/dependency).
+
+## Exporting different modules as a single XCFramework
+
+If your project contains unrelated KMP modules which you would like to export as a single iOS binary, you can create
+an umbrella module and combine other modules in it.
+
+For example, you have a `network` and a `database` module, which you combine in an `together` module.
```

Comment:
> `together` module

Do I recall correctly that the issue with using `umbrella` name for the module was that `umbrella` is a reserved identifier in the modulemap syntax? I wonder if we should deprecate the `umbrella` terminology in favor of something better

Reply:
I would rather leave an explicit note underlining that the user should not use `umbrella` as a name for the module.  It's a good name, that's why I used it :) and I have no doubt that a significant amount of people will think the same. Xcode says absolutely nothing when you do that, and there is no error except at the very end when the import just doesn't work, so we can save them a lot of trouble.

But I'll discuss this with the TW team.

Reply:
I think this issue is actually be quite easy to fix. I filed an issue - https://youtrack.jetbrains.com/issue/KT-66565